tools/site_admin/manage_special_days.php

```diff
@@ -127,21 +127,22 @@
 class SpecialDay
 {
     public bool $new_source;
+    // Note: spec_code cannot be null in the DB but this class makes it optional.
```

Comment:
By itself, I found this comment to be confusing. Since this isn't an autoincrement field, but requires `$spec_code` to be created by the person entering the new special day, the code has to make sure that the value is not null. Perhaps add something along the lines of
```php
// to ensure that the value in the form for the field is not null
```

Reply:
Sounds good. I have expanded the comment, which I think captured your suggestion. Let me know if that was what you had in mind:

```
// spec_code cannot be null in the DB. However to reuse the HTML form generation
// logic the first time the page is loaded, we do allow a null value in this class.
// Thus when reading the value from the form to store in the DB, we must ensure
// that it is not null.
```